app/controllers/mixins/generic_session_mixin.rb

```diff
@@ -13,6 +13,7 @@ def get_session_data
       @filters    = session["#{prefix}_filters".to_sym]
       @catinfo    = session["#{prefix}_catinfo".to_sym]
       @showtype   = session["#{prefix}_showtype".to_sym]
+      @current_page = session["#{prefix}_current_page".to_sym]
```

Comment:
You can actually just skip `@current_page` - remove it from `get_session_data` and `set_session_data` and *not* add them to the mixin.

That's because `get_global_session_data` and `set_global_session_data` already handle `@current_page` ([set](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/application_controller.rb#L2181) and [get](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/application_controller.rb#L1948)). (Then, `@sb` is saved in the session under `session[:sandboxes][controller_name]`.)